core/src/main/scala/com/rackspace/com/papi/components/checker/step/StepHandler.scala

```diff
@@ -755,7 +755,7 @@ class StepHandler(var contentHandler : ContentHandler, val config : Config) exte
       expression = XPathExpressionPool.borrowExpression(_match, context, version)
     } catch {
       case spe : SAXParseException => throw spe
-      case e : Exception => throw new SAXParseException ("Error while compiling XPath expression", locator, e)
+      case e : Exception => throw new SAXParseException ("Error while compiling XPath expression: "+e.getMessage(), locator, e)
```

Comment:
I noticed the exception was listing useful human readable information as to why things failed, when handling JSON XPath and display the message there. So extend to support regular XPath calls as well.